File: LeetCodeCPP/1732.find-the-highest-altitude.cpp
/*
 * @lc app=leetcode id=1732 lang=cpp
 *
 * [1732] Find the Highest Altitude
 */

#include <vector>

using namespace std;

// @lc code=start
class Solution
{
public:
    int largestAltitude(vector<int> &gain)
    {
        int maxAlt = 0;
        int currentAlt = 0;
        for (auto alt : gain)
        {
            currentAlt += alt;
            maxAlt = max(maxAlt, currentAlt);
        }
        return maxAlt;
    }
};
// @lc code=end

File: LeetCodePython/1732.find-the-highest-altitude.py
#
# @lc app=leetcode id=1732 lang=python3
#
# [1732] Find the Highest Altitude
#

from typing import List


# @lc code=start
class Solution:
    def largestAltitude(self, gain: List[int]) -> int:
        max_alt = 0
        current_alt = 0
        for alt in gain:
            current_alt += alt
            max_alt = max(max_alt, current_alt)
        return max_alt


# @lc code=end
